src/ccsr/src/client.rs

```diff
@@ -19,20 +20,29 @@ use serde::{Deserialize, Serialize};
 use crate::config::Auth;
 
 /// An API client for a Confluent-compatible schema registry.
-#[derive(Debug)]
 pub struct Client {
     inner: reqwest::Client,
-    url: Url,
+    url: Arc<dyn Fn() -> Url + Send + Sync + 'static>,
     auth: Option<Auth>,
 }
 
+impl fmt::Debug for Client {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Client")
+            .field("inner", &self.inner)
+            .field("url", &"...")
+            .field("auth", &self.auth)
+            .finish()
+    }
+}
+
 impl Client {
     pub(crate) fn new(
         inner: reqwest::Client,
-        url: Url,
+        url: Arc<dyn Fn() -> Url + Send + Sync + 'static>,
         auth: Option<Auth>,
     ) -> Result<Self, anyhow::Error> {
-        if url.cannot_be_a_base() {
+        if url().cannot_be_a_base() {
```

Comment:
nit, which I'll probably try to clean up later and def shouldn't block this PR: this validation is no longer worth all that much, because at any point later the closure can start returning a not-a-base URL.